httpserver/http_connection.cpp

```diff
@@ -160,9 +161,9 @@ void request_swarm_update(boost::asio::io_context& ioc, const swarm_callback_t&&
 
     make_http_request(
         ioc, ip, port, target, req_body,
-        [cb = std::move(cb)](const std::shared_ptr<std::string>& result_body) {
-            if (result_body) {
-                parse_swarm_update(result_body, std::move(cb));
+        [cb = std::move(cb)](const sn_response_t&& res) {
```

Comment:
`make_http_request` takes a `http_callback_t` which takes a `http_response_t` all by value, and that's fine since `sn_response_t` is an enum and a shared pointer, nothing heavy to copy.
You are already moving the response when calling the callback, so I guess the compiler will take care of optimising the rest and there is probably no need to use the rvalue ref `&&` here. Also, the other http request are happily taking `http_response_t` by value, so let's do this everywhere for consistency?

Reply:
It is actually not that cheap to copy compared to using `std::move`. Feels wrong to atomically increment the reference count only to decrement it almost immediately after.

Reply:
Actually, I'm not sure about the value not being copied, I had the example of std::unique_ptr in mind but that doesn't apply here.
But regardless, I think it's fine taking by value since the heavy data is already a shared_ptr.

Reply:
Ok, I agree with the ref count argument. So let's change the other callbacks to take by rvalue ref too then.

Reply:
Changed to use rvalue refs everywhere.